app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,53 @@
+<main class="flex flex-col content-center items-center">
+<%= render 'shared/menu' %>
+  <div class="header">
+    <h1 class="text-4xl font-extrabold dark:text-white m-5"><%= @recipe.name %></h1>  
+  </div>
+
+<div class="flex w-5/6 justify-between">
+  <div class="flex flex-col gap-5 font-semibold">
+    <p>Preparation time: <%= @recipe.preparation_time %> minutes</p>
+    <p>Cooking time: <%= @recipe.cooking_time %> minutes</p>
+    <p><%= @recipe.description %></p>
+  </div>
+    <%= form_for @recipe, url: toggle_public_recipe_path(@recipe), method: :patch do |f| %>
+    <label class="relative inline-flex items-center mr-5">
+      <%= f.check_box :public, class: "sr-only peer", checked: @recipe.public %>
+      <div class="w-11 h-6 bg-gray-200 rounded-full peer dark:bg-gray-700 peer-focus:ring-4 peer-focus:ring-green-300 dark:peer-focus:ring-green-800 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-0.5 after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-green-600"></div>
+      <%= f.submit "Public", class: "cursor-pointer" %>
+    </label> 
+  <% end %>  
+  </div>  
+  <% if @recipe.user == current_user %>
+    <a href = <%= new_recipe_recipe_food_path(@recipe.id) %> class="flex self-end mr-20">
+      <button class='bg-white hover:bg-gray-100 text-gray-800 font-semibold py-1 px-2 border border-gray-400 rounded shadow'>Add Food</button>
+    </a>
+  <% end %>
```

Comment:
- [ ] Please you are missing the `general_shopping_list` button which should be able to generate a shopping list. Also, the `add food` button here should be `Add ingredient` 

  - While adding the `general_shopping_list` button make sure you implement its functionality as well :+1: 

- [ ] I think recipes should be able to switch from public to private and remember that switch from public to private should not be found in public recipes. 

  [Screencast from 04-07-2023 03:32:34 PM.webm](https://user-images.githubusercontent.com/92303891/230635956-d05e6e42-88fe-4ddd-b0db-3bfcadc294eb.webm)

Reply:
@Owusu-Desmond Hi, recipes can switch from public to private by clicking on "Public"

Reply:
@Owusu-Desmond Also, the generate shopping list seems to be only for groups of 3 or more people https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#how-to-build-the-recipe-app